### ISSUE TITLE
BigInt constructor should not accept any type

### ISSUE THREAD

**Issue by LongTengDao** [microsoft/TypeScript#38980]
**TypeScript Version:**  3.9.3

**Search Terms:**

**Code**

**Expected behavior:**

```ts
interface BigIntConstructor {
    (value: string | number | bigint | boolean | object): bigint;
}
```

**Actual behavior:**

```ts
interface BigIntConstructor {
    (value?: any): bigint;
}
```

**Playground Link:** 

**Related Issues:**

**Comment by DanielRosenwasser** [comment]
I think a PR would be a good start to a discussion. We could then run it on a few code suites and see what breaks.

**Comment by idankash** [comment]
Hey,
I'd love to open a PR and work on it if it's ok. :)

**Comment by dy-fi** [comment]
I will submit a PR on this if it's still open

**Comment by LongTengDao** [comment]
> I will submit a PR on this if it's still open

There is already #38995. Don't know why it's still not merged.

**Comment by KisaragiEffective** [comment]
maybe fixed by #43204
